tests/test_image_vault.cpp

```diff
@@ -420,3 +448,19 @@ TEST_F(ImageVault, valid_remote_and_alias_returns_valid_image_info)
     EXPECT_THAT(image.original_release, Eq("16.04 LTS"));
     EXPECT_THAT(image.id, Eq("e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855"));
 }
+
+TEST_F(ImageVault, http_download_returns_expected_image_info)
+{
+    HttpURLDownloader http_url_downloader;
+    mp::DefaultVMImageVault vault{hosts, &http_url_downloader, cache_dir.path(), data_dir.path(), mp::days{0}};
+
+    auto image_url{"http://www.foo.com/images/foo.img"};
+    mp::Query query{instance_name, image_url, false, "", mp::Query::Type::HttpDownload};
+
+    mp::VMImage image;
+    EXPECT_NO_THROW(image = vault.fetch_image(mp::FetchType::ImageOnly, query, stub_prepare, stub_monitor));
```

Comment:
Was a throw the cause of the crash? Just for my understanding.

Reply:
No, it wasn't the cause of the crash.  Just putting the `EXPECT_NO_THROW` because, well, we don't expect a throw here :smiley:

Reply:
OK, fine by me